signer/xpi/xpi.go

```diff
@@ -303,6 +323,42 @@ func (s *XPISigner) signDataWithPKCS7(sigfile []byte, cn string, digest asn1.Obj
 	return p7sig, nil
 }
 
+// SendGauge checks for a client and when one is present sends a
+// statsd gauge with the given name, int value cast to
+// float64, tags for the signer, and sampling rate of 1
+func (s *XPISigner) SendGauge(name string, value int) {
+	if s.stats == nil {
+		log.Warnf("xpi: statsd client is nil. Could not send gauge %s with value %v", name, value)
+		return
+	}
+	var (
+		err        error
+		signerTags []string = []string{s.ID, s.Type, s.Mode}
+	)
+	err = s.stats.Gauge(name, float64(value), signerTags, 1)
+	if err != nil {
+		log.Warnf("Error sending gauge %s: %s", name, err)
+	}
+}
+
+// SendHistogram checks for a client and when one is present sends a
+// statsd histogram with the given name, time.Duration value cast to
+// float64, tags for the signer, and sampling rate of 1
+func (s *XPISigner) SendHistogram(name string, value time.Duration) {
+	if s.stats == nil {
+		log.Warnf("xpi: statsd client is nil. Could not send histogram %s with value %s", name, value)
+		return
+	}
+	var (
+		err        error
+		signerTags []string = []string{s.ID, s.Type, s.Mode}
+	)
+	err = s.stats.Histogram(name, float64(value), signerTags, 1)
+	if err != nil {
+		log.Warnf("Error sending histogram %s: %s", name, err)
+	}
+}
```

Comment:
I can think of other uses for these functions outside of the xpi signer. Would it be useful to export them from the `go.mozilla.org/autograph/signer` package?

Reply:
Yeah I'd like to add the signer ID, type, and mode tags to all stats we send from signers.

I think we can wrap the datadog statsd client and export from the signer pkg.

I was also have a branch that adds the cache as autograph-shared resource, but we end up specifying key sizes which might be worse.